flink-architecture-tests/flink-architecture-tests-base/src/main/java/org/apache/flink/architecture/common/Predicates.java

```diff
@@ -62,40 +68,42 @@ public static DescribedPredicate<JavaClass> containAnyFieldsInClassHierarchyThat
     }
 
     /**
-     * Tests that the given field is {@code public static} and of the given type {@code clazz} .
+     * Tests that the given field is {@code public static} and has the fully qualified type name of
+     * {@code fqClassName}.
      *
      * <p>Attention: changing the description will add a rule into the stored.rules.
      */
-    public static DescribedPredicate<JavaField> arePublicStaticOfType(Class<?> clazz) {
```

Comment:
Do we know whether any of these methods are also used by connector repos? Wondering about compile error when working against newer Flink versions.

Reply:
Good point! I ran a script to search for the following methods
```
areFieldOfType
arePublicFinalOfType
arePublicStaticOfType
arePublicStaticFinalOfType
arePublicFinalOfTypeWithAnnotation
arePublicStaticFinalOfTypeWithAnnotation
areStaticFinalOfTypeWithAnnotation
```
across these repositories
```
apache/flink-connector-kafka
apache/flink-connector-cassandra
apache/flink-connector-elasticsearch
apache/flink-connector-rabbitmq
apache/flink-connector-google-cloud-pubsub
apache/flink-connector-pulsar
apache/flink-connector-jdbc
apache/flink-connector-hbase
apache/flink-connector-hive
apache/flink-connector-aws
apache/flink-connector-firehose
apache/flink-connector-kinesis
apache/flink-connector-dynamodb
```
and did not find any usage.

It seems that the question "Does this change in Flink break any connector?" will be a recurring topic with the externalized repos. Having some automated checks is tricky, but feasible:
1) get all the method signature changes in the PR
2) search across the connector repos for usage of the respective classes + methods
The experience I made is that it won't work with GitHub API due to pretty heavy throttling on their side. What do, will  checking out all the connector repos in the main Flink CI run work for us?

Reply:
> What do think, could checking out all the connector repos in the main Flink CI run work for us?

I think we just to clarify what is part of the user (connector?) facing API, annotate them accordingly, and them treat them as other public APIs.